### PR TITLE
make piecharts clickable

### PR DESCRIPTION
**Pull request by rayene**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-go-api#383.0 ✘  
* Build sources
  * Build API ✔ 
* Integration tests
  * Integration tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-go-api#384.0 ✘  
* Build sources
  * Build API ✔ 
* Integration tests
  * Integration tests ✘

**Comment by ovh-cds** [comment]
CDS Report deploy-preview#386.0 ✘  
* 
  * deploy-platform ✘

File: api/v1/application/repository_latest.go [hunk 1]
```diff
@@ -106,8 +107,21 @@ func getWhereClause(criteria map[string]interface{}) string {
 					OR cast("av".manifest as TEXT) ILIKE '%%%s%%'
 					OR cast("av".tags as TEXT) ILIKE '%%%s%%'
 				) `, value, value, value, value, value, value)
+		} else if strings.Contains(key, ".") {
```
Comment (FabienM) [inline api/v1/application/repository_latest.go]: Cyclomatic and N-path complexities are too high:
```
if {
} else if {
  for {
   if {
   } elseif {
   } else {
   }
  }
} else {
}
```

This should be refactored with submethods to keep only one level, avoiding `else if` (and `else`, as much as possible).

File: api/v1/application/repository_latest.go [hunk 2]
```diff
@@ -97,7 +98,7 @@ func getWhereClause(criteria map[string]interface{}) string {
 			continue
 		}
 		// TODO SQL injection
-		if key == "search" {
+		if key == "freesearch" {
```
Comment (FabienM) [inline api/v1/application/repository_latest.go]: This whole algorithm could be genenalized in `hateoas/utils.go` or something in order to be reusable in other repositories.

Comment (FabienM) [inline api/v1/application/repository_latest.go]: `freesearch` may be a constant in `hateoas/type.go`?

**Comment by ovh-cds** [comment]
CDS Report build-go-api#386.1 ✘  
* Build sources
  * Build API ✔ 
* Integration tests
  * Integration tests ✘

File: api/v1/application/repository_latest.go (hunk 2 above)
Comment (rayene) [reply api/v1/application/repository_latest.go]: I agree. However, I do not see an easy way to do it right now without a big refactoring. What about doing it outside of this PR ??

File: api/v1/type.go [hunk 3]
```diff
@@ -29,6 +29,8 @@ const ContentBasePath = "/content"
 // BadgeBasePath is the URL base path for this resource
 const BadgeBasePath = "/badges"
 
+const FreeSearch = "freesearch"
```
Comment (FabienM) [inline api/v1/type.go]: missing comment

**Comment by ovh-cds** [comment]
CDS Report build-go-api#396.0 ✘  
* Build sources
  * Build API ✔ 
* Integration tests
  * Integration tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-go-api#396.1 ✘  
* Build sources
  * Build API ✔ 
* Integration tests
  * Integration tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-go-api#396.2 ✘  
* Build sources
  * Build API ✔ 
* Integration tests
  * Integration tests ✘

**Pull request merged**